GPUSolver/gpu/bse_explicit_parallel.cu

```diff
@@ -0,0 +1,129 @@
+//
+// Created by Chelsea De Marseilla on 07/02/2025.
+//
+
+#include <iostream>
+#include <chrono>
+// #include "include/tensor.cuh" // how do i link GPUtils to PDESolvers???
+
+enum class OptionType
+{
+    Call,
+    Put
+};
+
+__global__ void set_terminal_condition(double* dev_grid, OptionType type, int s_nodes, int t_nodes, double dS,
+                                       double strike_price)
+{
+    using enum OptionType;
+    int idx = threadIdx.x + blockIdx.x * blockDim.x;
+
+    int index = (t_nodes) * (s_nodes + 1) + idx;
+
+    if (idx < s_nodes + 1)
+    {
+        if (type == Call)
```

Comment:
After you change this using template parameters, change this to `if constexpr (type == Call)`